### DISCUSSION TITLE
Question: This typedef is deprecated in favor of haxe.xml.Access

### DISCUSSION THREAD

**Discussion by TomiokaGiyuGamer** [FunkinCrew/Funkin#1525] [Compiling help]
## What is your question?
![imagem_2021-04-15_222031](https://user-images.githubusercontent.com/68976448/114958125-caa37800-9e38-11eb-892d-191ae4a256d8.png)

**Comment by ghost** [comment]
so the reason it appears it doesnt work is because you didnt make the APIStuff.hx file. the polymod and haxe xml access shit doesnt matter, because it happens to me and still works.

in the "source" folder make a txt file and call it "APIStuff.hx" make sure the file type is hx and not txt.
copy and paste this into it

package;

class APIStuff
{
	public static var API:String = "";
	public static var EncKey:String = "";
}

and u should be able to compile it!

**Reply by ghost** [reply]
i already have this  (im italian)
![Cattura](https://user-images.githubusercontent.com/87648887/131226020-66bca600-5c05-483c-9f37-4c66880e85cc.PNG)

**Comment by foxsouns** [comment]
> package;
> 
> class APIStuff
> {
> public static var API:String = "";
> public static var EncKey:String = "";
> }

this dont work for me, i need to use the one in https://github.com/ninjamuffin99/Funkin/issues/146#issuecomment-755064854

**Comment by SinonsAnimations** [comment]
> so the reason it appears it doesnt work is because you didnt make the APIStuff.hx file. the polymod and haxe xml access shit doesnt matter, because it happens to me and still works.
> 
> in the "source" folder make a txt file and call it "APIStuff.hx" make sure the file type is hx and not txt.
> copy and paste this into it
> 
> package;
> 
> class APIStuff
> {
> public static var API:String = "";
> public static var EncKey:String = "";
> }
> 
> and u should be able to compile it!

dont work i need other plis

**Comment by foxsouns** [comment]
> dont work i need other plis

look at the link i have in my comment, it's some persons dummy APIStuff.hx

**Comment by ghost** [comment]
The "This typedef" message does not  stop the game compiling.

**Comment by AshuraTheEchidna** [comment]
> > dont work i need other plis
> 
> look at the link i have in my comment, it's some persons dummy APIStuff.hx

it still didnt worked for me

**Comment by steviegt6** [comment]
> > > dont work i need other plis
> > 
> > 
> > look at the link i have in my comment, it's some persons dummy APIStuff.hx
> 
> it still didnt worked for me

The missing `APIStuff` class is unrelated (though this is the reason the original OP wasn't able to compile). This is a warning from `polymod`, not `Fukin`.

**Comment by AshuraTheEchidna** [comment]
> > > > dont work i need other plis
> > > 
> > > 
> > > look at the link i have in my comment, it's some persons dummy APIStuff.hx
> > 
> > 
> > it still didnt worked for me
> 
> The missing `APIStuff` class is unrelated (though this is the reason the original OP wasn't able to compile). This is a warning from `polymod`, not `Fukin`.

ok but there is a way to fix it? because my game is not compiling because of that

**Comment by steviegt6** [comment]
> > > > > dont work i need other plis
> > > > 
> > > > 
> > > > look at the link i have in my comment, it's some persons dummy APIStuff.hx
> > > 
> > > 
> > > it still didnt worked for me
> > 
> > 
> > The missing `APIStuff` class is unrelated (though this is the reason the original OP wasn't able to compile). This is a warning from `polymod`, not `Fukin`.
> 
> ok but there is a way to fix it? because my game is not compiling because of that

As per the README:
![image](https://user-images.githubusercontent.com/27323911/118844721-0b4f4080-b880-11eb-928d-352f91780cf5.png)

**Comment by AshuraTheEchidna** [comment]
> > > > > > dont work i need other plis
> > > > > 
> > > > > 
> > > > > look at the link i have in my comment, it's some persons dummy APIStuff.hx
> > > > 
> > > > 
> > > > it still didnt worked for me
> > > 
> > > 
> > > The missing `APIStuff` class is unrelated (though this is the reason the original OP wasn't able to compile). This is a warning from `polymod`, not `Fukin`.
> > 
> > 
> > ok but there is a way to fix it? because my game is not compiling because of that
> 
> As per the README:
> ![image](https://user-images.githubusercontent.com/27323911/118844721-0b4f4080-b880-11eb-928d-352f91780cf5.png)

source/PlayState.hx:298: characters 3-36 : Cannot rebind this method : please use 'dynamic' before method declaration
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access

**it still gives me the same error**

**Comment by ghost** [comment]
```
source/PlayState.hx:298: characters 3-36 : Cannot rebind this method : please use 'dynamic' before method declaration
```
This is the error that's stopping you from compiling. Look for lines _other than the XML access warnings_

**Comment by steviegt6** [comment]
> > > > > > > dont work i need other plis
> > > > > > 
> > > > > > 
> > > > > > look at the link i have in my comment, it's some persons dummy APIStuff.hx
> > > > > 
> > > > > 
> > > > > it still didnt worked for me
> > > > 
> > > > 
> > > > The missing `APIStuff` class is unrelated (though this is the reason the original OP wasn't able to compile). This is a warning from `polymod`, not `Fukin`.
> > > 
> > > 
> > > ok but there is a way to fix it? because my game is not compiling because of that
> > 
> > 
> > As per the README:
> > ![image](https://user-images.githubusercontent.com/27323911/118844721-0b4f4080-b880-11eb-928d-352f91780cf5.png)
> 
> source/PlayState.hx:298: characters 3-36 : Cannot rebind this method : please use 'dynamic' before method declaration
> C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
> C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
> 
> **it still gives me the same error**

Your error is unrelated to the issue created by the OP.

**Comment by ghost** [comment]
well i have APIStuff.hx but its shows me like that C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access

**Comment by somefnfplayerlol** [comment]
> C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access

this error does not stop the compiling. its just a warning

**Comment by KitOdin** [comment]
first off ; those are warnings, ignore them.
secondly : make a apistuff.hx file you dumbass
